pkg/agent/cniserver/container_configuration_unix.go

```diff
@@ -0,0 +1,356 @@
+	if err != nil {
+		return nil, err
+	}
+
+	linkName := link.Attrs().Name
+	veth := &vethPair{}
+	_, veth.peerIndex, err = ip.GetVethPeerIfindex(linkName)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get veth peer index for veth %s: %v", linkName, err)
+	}
+	veth.ifIndex = link.Attrs().Index
+	if intf.Mac != link.Attrs().HardwareAddr.String() {
+		return nil, fmt.Errorf("interface %s MAC %s doesn't match container MAC: %s",
+			intf.Name, intf.Mac, link.Attrs().HardwareAddr.String())
+	}
+	veth.name = linkName
+	return veth, nil
+}
+
+func (pc *podConfigurator) validateContainerPeerInterface(interfaces []*current.Interface, containerVeth *vethPair) (*vethPair, error) {
+	// Iterate all the passed interfaces and look up the host interface by
+	// matching the veth peer interface index.
+	for _, hostIntf := range interfaces {
+		if hostIntf.Sandbox != "" {
+			// Not in the default Namespace. Must be the container interface.
+			continue
+		}
+		link, err := validateInterface(hostIntf, false)
+		if err != nil {
+			klog.Errorf("Failed to validate interface %s: %v",
+				hostIntf.Name, err)
+			continue
+		}
+
+		if link.Attrs().Index != containerVeth.peerIndex {
+			continue
+		}
+
+		hostVeth := &vethPair{ifIndex: link.Attrs().Index, name: link.Attrs().Name}
+		_, hostVeth.peerIndex, err = ip.GetVethPeerIfindex(hostVeth.name)
+		if err != nil {
+			return nil, fmt.Errorf("failed to get veth peer index for host interface %s: %v",
+				hostIntf.Name, err)
+		}
+
+		if hostVeth.peerIndex != containerVeth.ifIndex {
+			return nil, fmt.Errorf("host interface %s peer index doesn't match container interface %s index",
+				hostIntf.Name, containerVeth.name)
+		}
+
+		if hostIntf.Mac != "" {
+			if hostIntf.Mac != link.Attrs().HardwareAddr.String() {
+				klog.Errorf("Host interface %s MAC %s doesn't match link address %s",
+					hostIntf.Name, hostIntf.Mac, link.Attrs().HardwareAddr.String())
+				return nil, fmt.Errorf("host interface %s MAC %s doesn't match",
+					hostIntf.Name, hostIntf.Mac)
+			}
+		}
+		return hostVeth, nil
+
+	}
+
+	return nil, fmt.Errorf("peer veth interface not found for container interface %s",
+		containerVeth.name)
+}
+
+func validateInterface(intf *current.Interface, inNetns bool) (netlink.Link, error) {
+	if intf.Name == "" {
+		return nil, fmt.Errorf("interface name is missing")
+	}
+	if inNetns {
+		if intf.Sandbox == "" {
+			return nil, fmt.Errorf("interface %s is expected in netns", intf.Name)
+		}
+	} else {
+		if intf.Sandbox != "" {
+			return nil, fmt.Errorf("interface %s is expected not in netns", intf.Name)
+		}
+	}
+
+	link, err := netlink.LinkByName(intf.Name)
+	if err != nil {
+		return nil, fmt.Errorf("failed to find link for interface %s", intf.Name)
+	}
+
+	_, isVeth := link.(*netlink.Veth)
+	if !isVeth {
+		return nil, fmt.Errorf("interface %s is not of type veth", intf.Name)
+	}
+	return link, nil
+}
+
+func (pc *podConfigurator) setupContainerOVSPort(
```

Comment:
I think we should share this func between Windows and Linux. Just extract createOVSPort() out.